src/operators/transfermatrix.jl

```diff
@@ -0,0 +1,250 @@
+import MPSKit: Multiline, MultilineEnvironments
+
+# TODO: rewrite in terms of InfiniteMPO{O} where O is an 'effective MPO type' once type
+# restrictions in MPSKit are relaxed
+
+#
+# PEPS
+#
+
+"""
+    InfiniteTransferPEPS{T}
+
+Represents an infinite transfer operator corresponding to a single row of a partition
+function which corresponds to the overlap between 'ket' and 'bra' `InfinitePEPS` states.
+"""
+struct InfiniteTransferPEPS{T}
+    top::PeriodicArray{T,1}
+    bot::PeriodicArray{T,1}
+    function InfiniteTransferPEPS(
+        top::PeriodicArray{T,1}, bot::PeriodicArray{T,1}
+    ) where {T}
+        size(top) == size(bot) ||
+            throw(ArgumentError("Top and bottom PEPS rows should have equal sizes."))
+        return new{T}(top, bot)
+    end
+end
+
+InfiniteTransferPEPS(top) = InfiniteTransferPEPS(top, top)
+
+"""
+    InfiniteTransferPEPS(T::InfinitePEPS, dir, row)
+
+Constructs a transfer operator corresponding to a single row of a partition function
+representing the norm of the state `T`. The partition function is first rotated such that
+the direction `dir` faces north, after which its `row`th row from the north is selected.
+"""
+function InfiniteTransferPEPS(T::InfinitePEPS, dir, row)
+    T = rotate_north(T, dir)
+    return InfiniteTransferPEPS(PeriodicArray(T[row, :]))
+end
+
+Base.size(transfer::InfiniteTransferPEPS) = size(transfer.top)
+Base.size(transfer::InfiniteTransferPEPS, args...) = size(transfer.top, args...)
+Base.length(transfer::InfiniteTransferPEPS) = size(transfer, 1)
+Base.getindex(O::InfiniteTransferPEPS, i) = (O.top[i], O.bot[i])
+
+Base.iterate(O::InfiniteTransferPEPS, i=1) = i > length(O) ? nothing : (O[i], i + 1)
```

Comment:
```suggestion
Base.iterate(O::InfiniteTransferPEPS, args...) = iterate(zip(O.top, O.bot), args...)
```
(Just a suggestion, can safely be ignored)